File: src/topic/src/person_listener.cpp
#include "ros/ros.h"
#include "topic/Person.h"

void doPerson(const topic::Person::ConstPtr& person_p){
    ROS_INFO("订阅的人信息:%s, %d, %.2f", person_p->name.c_str(), person_p->age, person_p->height);
}

int main(int argc, char *argv[])
{   
    setlocale(LC_ALL,"");

    ros::init(argc,argv,"listener_person");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe<topic::Person>("chatter_person",10,doPerson);

    ros::spin();    
    return 0;
}

File: src/topic/scripts/person_listener.py

import rospy
from topic.msg import Person

def doPerson(p):
    rospy.loginfo("接收到的人的信息:%s, %d, %.2f",p.name, p.age, p.height)


if __name__ == "__main__":
    rospy.init_node("listener_person_p")
    sub = rospy.Subscriber("chatter_person",Person,doPerson,queue_size=10)
    rospy.spin() 